cachi2/core/package_managers/generic/main.py

```diff
@@ -45,34 +52,50 @@ def _resolve_generic_lockfile(source_dir: RootedPath, output_dir: RootedPath) ->
             ),
         )
 
+    # output_dir is now the root and cannot be escaped
+    output_dir = output_dir.re_root(DEFAULT_DEPS_DIR)
+
     log.info(f"Reading generic lockfile: {lockfile_path}")
-    lockfile = _load_lockfile(lockfile_path)
+    lockfile = _load_lockfile(lockfile_path, output_dir)
+    to_download: dict[str, Union[str, os.PathLike[str]]] = {}
+
+    for artifact in lockfile.artifacts:
+        # create the parent directory for the artifact
+        Path.mkdir(Path(artifact.target).parent, parents=True, exist_ok=True)
+        to_download[str(artifact.download_url)] = artifact.target
+
+    asyncio.run(async_download_files(to_download, get_config().concurrency_limit))
+
+    # verify checksums
     for artifact in lockfile.artifacts:
-        log.debug(f"Resolving artifact: {artifact.download_url}")
+        must_match_any_checksum(artifact.target, artifact.formatted_checksums)
     return []
 
 
-def _load_lockfile(lockfile_path: RootedPath) -> GenericLockfileV1:
+def _load_lockfile(lockfile_path: RootedPath, output_dir: RootedPath) -> GenericLockfileV1:
     """
     Load the cachi2 generic lockfile from the given path.
 
     :param lockfile_path: the path to the lockfile
+    :param output_dir: path to output directory
     """
     with open(lockfile_path, "r") as f:
         try:
             lockfile_data = yaml.safe_load(f)
         except yaml.YAMLError as e:
-            raise PackageRejected(
```

Comment:
You revert this hunk in the last commit. Sounds like a rebase typo.

Reply:
Yes, I messed that up during splitting of the commits, I meant to edit the below error. Let me fix that.

Reply:
Fixed. The changes related to PackageManagerError -> PackageRejected are now self contained in 6520893cf5d2cab5650f227d84a91e6b0fc5b5f5 and is the first commit in this PR.